database/scripts/seed.ts

```diff
@@ -0,0 +1,191 @@
+/* eslint-disable @typescript-eslint/no-var-requires */
+require("dotenv").config({ path: ".env.local" });
+const MongoClient = require("mongodb").MongoClient;
+const faker = require("faker");
+
+// seed config
+const QUESTION_COUNT = 24;
+const GROUP_COUNT = 5;
+const COMPONENT_COUNT = 64;
+const SLIDE_COUNT = 16;
+const MODULE_COUNT = 3;
+const TOOL_COUNT = 2;
+
+const QUESTIONS_PER_GROUP = Math.floor(QUESTION_COUNT / GROUP_COUNT);
+const COMPONENTS_PER_SLIDE = Math.floor(COMPONENT_COUNT / SLIDE_COUNT);
+const SLIDES_PER_MODULE = Math.floor(SLIDE_COUNT / MODULE_COUNT);
+
+(async function () {
+    const client = new MongoClient(process.env.MONGODB_URI, {
+        useNewUrlParser: true,
+        useUnifiedTopology: true,
+    });
+
+    try {
+        await client.connect();
+        console.log("Successfully connected to database");
+
+        const db = client.db(process.env.MONGODB_DB);
+        const questionCollection = db.collection("SelfCheckQuestion");
+        const groupCollection = db.collection("SelfCheckGroup");
+        const componentCollection = db.collection("Component");
+        const slideCollection = db.collection("Slide");
+        const moduleCollection = db.collection("Module");
+        const toolCollection = db.collection("Tool");
+
+        if (process.argv[2] != "--init") {
+            // destroy previous data
+            await Promise.all([
+                questionCollection.drop(),
+                groupCollection.drop(),
+                componentCollection.drop(),
+                slideCollection.drop(),
+                moduleCollection.drop(),
+                toolCollection.drop(),
+            ]);
+        }
+
+        await Promise.all([
+            questionCollection.insertMany(mockQuestions()),
+            groupCollection.insertMany(mockGroups()),
+            componentCollection.insertMany(mockComponents()),
+            slideCollection.insertMany(mockSlides()),
+            moduleCollection.insertMany(mockModules()),
+            toolCollection.insertMany(mockTools()),
+        ]);
+
+        console.log("Successfully completed seeding");
+        client.close();
+    } catch (err) {
+        console.log(err.stack);
+    }
+})(); // immediately-invoked function expression
+
+function mockQuestions() {
+    const questionTypes = ["multiple_choice", "multi_select"];
+    const mockOptions = [
+        [1, 2, 3],
+        ["yes", "no", "maybe"],
+    ];
+    const questions = [];
+
+    for (let i = 0; i < QUESTION_COUNT; i++) {
+        questions.push({
+            _id: i,
+            type: questionTypes[i % questionTypes.length],
+            question: faker.lorem.words() + faker.lorem.words() + "?",
+            options: mockOptions[i % mockOptions.length],
+            questionNumber: (i % QUESTIONS_PER_GROUP) + 1,
+        });
+    }
+    return questions;
+}
+
+function mockGroups() {
+    const groups = [];
```

Comment:
is there a reason we don't set the `_id` for a group here?

Reply:
good catch! we'll need to set `_id` so we can link these groups with tools